### DISCUSSION TITLE
afterfolder hook and flatpak

### DISCUSSION THREAD

**Discussion by boyska** [nicotine-plus/nicotine-plus#2430] [Q&A]
Hi,
I'm running nicotine-plus with the flatpak from flathub. It works great! thanks for that.

However, the `afterfolder` hook doesn't work. I understand that the problem is that, inside a flatpak, not every path is accessible. What kind of workaround do you have?

By the way, what I want to do is very simple: move a folder to another directory. I'd like to have fully-completed dirs in one place, while still-downloading folders are in another place. This would allow me for easier batch importing.

**Comment by slook** [comment]
> What kind of workaround do you have?

You could write a plugin that uses the `download_finished_notification()` hook.

Alternatively perhaps a bash script might be able to elevate over whatever path restrictions that are encountered on your system.

In the future it is intended to move the built-in after hooks from the core into a plugin, because there are security and permissions related problems with the current implementation as you have highlighted.
